### PR TITLE
[Icds dashboard] add person case field

### PR DESCRIPTION
**Pull request by Rohit25negi**
First commit adds the essential fields important for the V1 of the Bihar Api.
Second commit adds the fields which will be needed at later stage. I think its better if we can fill all fields together to prevent rebuilding the data source again.

ticket: https://app.asana.com/0/1112385193248823/1160919599791898/f

### CODE REVIEW

File: custom/icds_reports/ucr/data_sources/person_cases_v3.json [hunk 1]
```diff
@@ -307,6 +307,248 @@
...
+          }
+        }
+      },
+      {
+        "column_id": "last_class_attended_ever",
+        "datatype": "integer",
+        "type": "raw",
+        "property_name": "last_class_attended_ever"
+      },
+      {
```
Comment (calellowitz) [inline custom/icds_reports/ucr/data_sources/person_cases_v3.json]: This is also ccs record case property that is already stored in the ccs record UCR, and mostly only makes sense in the context of active pregnancies. What will this be used for? I could not find it in the spec, but if it is in the pregnancy part of the API, the ccs record one is probably fine

File: custom/icds_reports/ucr/data_sources/person_cases_v3.json [hunk 2]
```diff
@@ -307,6 +307,248 @@
...
+            }
+          },
+          "default": {
+            "type": "constant",
+            "constant": null
+          }
+        }
+      },
+      {
+        "column_id": "last_class_attended_ever",
```
Comment (calellowitz) [inline custom/icds_reports/ucr/data_sources/person_cases_v3.json]: what is this indicator?

**Review by calellowitz — COMMENTED** [review]
One concern, one question for curiosity, but otherwise looks good assuming these are the right case properties. Good call adding future ones

File: custom/icds_reports/ucr/data_sources/person_cases_v3.json (hunk 2 above)
Comment (Rohit25negi) [reply custom/icds_reports/ucr/data_sources/person_cases_v3.json]: this "LastClassAttended" , Below is the comment on this in the app.
```
The last class attended by the girl: If first time registration (was_oos_ever_loaded!=yes) then take from register_ag group ; if re-registration then take from re-register_ag group ; else check what is autoloaded.

```

File: custom/icds_reports/ucr/data_sources/person_cases_v3.json (hunk 1 above)
Comment (Rohit25negi) [reply custom/icds_reports/ucr/data_sources/person_cases_v3.json]: Actually this not mentioned in any category in the first tab of the sheet. Not sure, where will this fit.
is ccs one  preg_order same as person case one?

Comment (Rohit25negi) [reply custom/icds_reports/ucr/data_sources/person_cases_v3.json]: I have asked Zohaib for confirmation. But it sounds like they should be same. If Zohaib does not respond by the EOD, I will remove ti.

File: custom/icds_reports/ucr/data_sources/person_cases_v3.json (hunk 2 above)
Comment (Rohit25negi) [reply custom/icds_reports/ucr/data_sources/person_cases_v3.json]: yeah, He confirmed we can use ccs one. removing ths.

File: custom/icds_reports/ucr/data_sources/person_cases_v3.json (hunk 1 above)
Comment (Rohit25negi) [reply custom/icds_reports/ucr/data_sources/person_cases_v3.json]: yes, he confirmed that we can use ccs one. removing this

**Comment by Rohit25negi** [comment]
@calellowitz  can you review this. if this looks good to you. I have answered the comments. I am planning to deploy this today.

**Review by calellowitz — APPROVED** [review]

**Pull request merged**